libs/core/langchain_core/document_stores.py

```diff
@@ -0,0 +1,553 @@
+"""**Document store** stores documents and performs similarity search."""
+from __future__ import annotations
+
+import warnings
+from abc import ABC, abstractmethod
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    ClassVar,
+    Collection,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+)
+
+from langchain_core.pydantic_v1 import Field, root_validator
+from langchain_core.retrievers import BaseRetriever
+from langchain_core.runnables.config import run_in_executor
+
+if TYPE_CHECKING:
+    from langchain_core.callbacks.manager import (
+        AsyncCallbackManagerForRetrieverRun,
+        CallbackManagerForRetrieverRun,
+    )
+    from langchain_core.documents import Document
+
+
+class DocumentStore(ABC):
+    """Interface for document store."""
+
+    @abstractmethod
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
```

Comment:
should this also have a list of optional ids, to match the delete method?

Reply:
This PR is about moving the method to a super class.
If modifications to the API are to be done, I think they should be done in another PR.
Also, since not all vector stores support passing explicit IDs when inserting documents, I'm not sure it is a good thing to add it to the core API.